### PR TITLE
fix(virgo): auto scroll to the focus text

### PR DESCRIPTION
**Pull request by Saul-Mirone**
Closes: #1655

Auto scroll when focus is out of viewport.

https://user-images.githubusercontent.com/10047788/224636680-b9ddbd34-b288-4163-8c5c-8f509f433087.mov

### CODE REVIEW

**Review by doodlewind — APPROVED** [review]

File: packages/virgo/src/virgo.ts [hunk 1]
```diff
@@ -714,6 +714,11 @@ export class VEditor<
     const range = document.createRange();
     range.setStart(anchorText, anchorOffset);
     range.setEnd(focusText, focusOffset);
+
+    if (scrollIntoView) {
+      focusText.parentElement?.scrollIntoView();
```
[the remark this answers is not in the document]
Comment (Saul-Mirone) [reply packages/virgo/src/virgo.ts:719]: Should we use it in product? Seems it's not a standard API and Firefox won't support it.

[the remark this answers is not in the document]
Comment (Saul-Mirone) [reply packages/virgo/src/virgo.ts:719]: https://github.com/scroll-into-view/scroll-into-view-if-needed
Maybe we can use this package.

[the remark this answers is not in the document]
Comment (lawvs) [reply packages/virgo/src/virgo.ts:719]: If this package can also solve the issue https://github.com/toeverything/blocksuite/issues/968, I would be more than happy to use it.

[the remark this answers is not in the document]
Comment (Saul-Mirone) [reply packages/virgo/src/virgo.ts:719]: Ok, I'll import it.

**Comment by lawvs** [comment]
I think the current scrolling behavior is not stable enough.

**Comment by Saul-Mirone** [comment]
> I think the current scrolling behavior is not stable enough.

Can you give it some details?

**Comment by lawvs** [comment]
> > I think the current scrolling behavior is not stable enough.
> 
> Can you give some details?

When there is content below the page, pressing Enter will center the next block, which I think is unexpected.

/cc @joebeijing 


https://user-images.githubusercontent.com/18554747/224706250-c0422412-f927-4723-a004-436a8043ccb8.mov

**Comment by Saul-Mirone** [comment]
> > > I think the current scrolling behavior is not stable enough.
> > 
> > 
> > Can you give some details?
> 
> When there is content below the page, pressing Enter will center the next block, which I think is unexpected.
> 
> /cc @joebeijing

Sounds it more like a feature strategy problem instead of technical detail. It's hard to say which is right. If I'm a user, I'd like to see the scroll behavior if the cursor disappear from my viewport, which probably means I can't find it so I just press `Enter`.

**Pull request merged**